### ISSUE TITLE
"Space" doesn't expand anymore

### ISSUE THREAD

**Issue by Jopp-gh** [ArianaKhit/text-snippets-obsidian#47]
Expansions aren't possible anymore using space or tab both on mobile and desktop devices. Older version of **text snippets**, allowed text expansion via "space" or "tab" key

Writing text needs the "Space" button as expander, everything else is cumbersome, because alternatives require 2 fingers. This issue is even worse on mobile! 

I'd argue that even the "tab" button is cumbersome because not so easy and quickly accessibile than the space bar button

**Comment by RobertGoulding** [comment]
I installed Text Snippets today, and found the same thing. Only Ctrl-Tab will expand, not Space or Tab. As the OP notes, Space is really necessary to auto-expand abbreviations as you type.

**Comment by neel-maniar** [comment]
yup, tab doesn't work for me either!

**Comment by tedmiston** [comment]
I tried this plugin on mobile with space expansion enabled and noticed that no expansion happened and wondered if it just didn't work on mobile at all.

Hopefully this bug explains that issue too.

**Comment by edofe99** [comment]
Any chance on getting this plugin actually working?

**Comment by FlynnD273** [comment]
I'm pretty sure as of the latest update, it _does_ work as intended.

**Comment by edofe99** [comment]
> I'm pretty sure as of the latest update, it _does_ work as intended.

https://user-images.githubusercontent.com/62611612/186849008-d8cb3bb8-1742-4fcf-839a-113e31b5c9a8.mov

...

**Comment by Oskiator** [comment]
Hey ! I just installed the plugin, but as mentionned above, space and tab doesn't work for me either. Only Ctrl+Tab.
It would be great also to have an option to autoexpand when writting some snippets. I make a separate feature request for that.

**Comment by FlynnD273** [comment]
You've enabled these switches in the plugin settings? It works perfectly for me.
![image](https://user-images.githubusercontent.com/30308842/186922675-0f49069f-332c-4a0d-8e3d-2bcbfc53fa39.png)

**Comment by edofe99** [comment]
> You've enabled these switches in the plugin settings? It works perfectly for me. ![image](https://user-images.githubusercontent.com/30308842/186922675-0f49069f-332c-4a0d-8e3d-2bcbfc53fa39.png)

Yes, even tried disabling all plugins and reloading app, nothing. 
Edit: I'm on macos last version of monterey running Obsidian v0.15.9

**Comment by FlynnD273** [comment]
I'm on Windows running 0.15.9. Do you have any other plugins? Perhaps a separate plugin is conflicting? Or even a theme? Try the default theme and disabling all other community plugins, perhaps?

**Comment by edofe99** [comment]
> I'm on Windows running 0.15.9. Do you have any other plugins? Perhaps a separate plugin is conflicting? Or even a theme? Try the default theme and disabling all other community plugins, perhaps?

Disabled all plugins and set on default theme, nothing, even on iPhone it won’t work. 

Perhaps if the creator of the plug-in is willing to investigate I’m aviable to share my vault for further investigations.

**Comment by mretier** [comment]
Same here, tried a completely new vault on windows, Obsidian v0.15.9

**Comment by Arax20** [comment]
It's not a plugin conflict, the issue is due to obsidian moving from CodeMirror 5 to 6 which breaks the function used to detect keypresses:

```
this.cmEditors = [];
this.registerCodeMirror((cm) => {
  this.cmEditors.push(cm);
  cm.on('keydown', (cm, event) => this.handleKeyDown(event));
});
```

SekoiaTree's [fork](https://github.com/SekoiaTree/text-snippets-obsidian/tree/patch-1) (make sure you're on the patch-1 branch) fixes this if you're willing to manually install the plugin.

**Comment by mretier** [comment]
I installed the SekoiaTree's fork. However, when testing the plugin in a new Vault in Obsidian 0.15.9 on Windows, Tab and Space still don't work.

**Comment by Arax20** [comment]
You have to run `npm run build` to rebuild from main.ts I think, main.js doesn't have the fix

**Comment by mretier** [comment]
@Arax20 I tried your suggestion, but receive the following error message (ran command in Windows powershell in the directory that contains the plugin files):


 npm run build

\> text-snippets-obsidian@0.1.1 build
\> rollup --config rollup.config.js

'rollup' is not recognized as an internal or external command,
operable program or batch file.

**Comment by edofe99** [comment]
Hi, thank's for temporary solution but I'll not do it because I'm not able. Do you know if admin will release an official update of this plugin? Thanks.

**Comment by jeansordes** [comment]
I've found a solution

I've built the file on my machine, and extracted the `main.js` file, you can get it below : https://gist.githubusercontent.com/jeansordes/9ccd17911d1743b4aaecd20a858d84e5/raw/57eafae03af2215a64591a23489de4419e527b62/main.js

This fils is meant to replace the following file : `.obsidian/plugins/text-snippets-obsidian/main.js`

This will do the job until the owner of this project accept the merge request

**Comment by RobertGoulding** [comment]
It works beautifully with that fix - thank you!

**Comment by cenosito** [comment]
> I've found a solution
> 
> I've built the file on my machine, and extracted the `main.js` file, you can get it below : https://gist.githubusercontent.com/jeansordes/9ccd17911d1743b4aaecd20a858d84e5/raw/57eafae03af2215a64591a23489de4419e527b62/main.js
> 
> This fils is meant to replace the following file : `.obsidian/plugins/text-snippets-obsidian/main.js`
> 
> This will do the job until the owner of this project accept the merge request

This works for me! Thanks

**Comment by greenwar** [comment]
> I've found a solution
> 
> I've built the file on my machine, and extracted the `main.js` file, you can get it below : https://gist.githubusercontent.com/jeansordes/9ccd17911d1743b4aaecd20a858d84e5/raw/57eafae03af2215a64591a23489de4419e527b62/main.js
> 
> This fils is meant to replace the following file : `.obsidian/plugins/text-snippets-obsidian/main.js`
> 
> This will do the job until the owner of this project accept the merge request

This worked! Plugin still broken until I patched with this. Thank you!

**Comment by jaysonteng** [comment]
> I've found a solution
> 
> I've built the file on my machine, and extracted the `main.js` file, you can get it below : https://gist.githubusercontent.com/jeansordes/9ccd17911d1743b4aaecd20a858d84e5/raw/57eafae03af2215a64591a23489de4419e527b62/main.js
> 
> This fils is meant to replace the following file : `.obsidian/plugins/text-snippets-obsidian/main.js`
> 
> This will do the job until the owner of this project accept the merge request

thank you very much, it works.

**Comment by alecbramham** [comment]
This hack doesn't work anymore (at least for me).
I've switched to using "Typing Transformer" from a different issue.
Very different syntax to this and ahk, but it gets the job done.

**Comment by Arax20** [comment]
@alecbramham I'd recommend you look at [obsidian-latex-suite](https://github.com/artisticat1/obsidian-latex-suite), which is what I've migrated to. It does the space bar functionality you're looking for with the auto expand snippets option.

Contrary to the name, it's not simply for writing latex - it's a very sophisticated and comprehensive snippets solution which allows for both manual and auto completion, regexing, scripted completions etc.

**Comment by jeansordes** [comment]
@Arax20 Oh, my, goodness. I just tried the plugin you mentionned : I love it 💖 Thank you very much dear stranger 🙏
